static/index.js

```diff
@@ -0,0 +1,82 @@
+let reviews = [];
+
+let sum = 0;
+
+if(localStorage.getItem("reviews") !== null) reviews = JSON.parse(localStorage.getItem("reviews"));
+
+
+
+const updateAverage = () =>{
+    document.querySelector("#averageRating").innerText = `${Math.round(sum / reviews.length * 10) / 10}/5`;
+}
+
+const createItem = (item) => {
+    let reviewItem = document.createElement("div");
+    let reviewText = document.createElement("p");
+    reviewText.innerText = item.review;
+    
+    let ratingText = document.createElement("p");
+    ratingText.innerText = `${item.rating} / 5`;
+
+    let deleteButton = document.createElement("button");
+    deleteButton.innerText = "Delete";
+    deleteButton.id = `item-${item.id}`;
+    
+    updateAverage();
+
+    deleteButton.addEventListener("click", (e)=>{
+        for(let rev in reviews){
+            console.log(`${e.target.id.split('-')[1] } VS..... ${reviews[rev].id}`)
+
+            if(e.target.id.split('-')[1] == reviews[rev].id) {
+                sum -= reviews[rev].rating;
```

Comment:
You are updating the global sum variable in a lot of places. This way you are never sure which piece of code last updated the sum. If you get a bug with the sum, it is very hard to identify the cause.
In my opinion, it would be much better to calculate the sum on the fly in the `updateAverage` function. The performance overhead for that is really minimal so there are really no drawbacks.

Reply:
Fixed in [Improvements](https://github.com/infinum-academy/js-react-academy-2024-filiptodosov/pull/1/commits/79575e604ff61d5206a96a781ad054151af9c333)